File: VB/WindowsApplication1/MyRadioListBox/MyRadioListBox.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory("")> _
	Public Class MyRadioListBox
		Inherits CheckedListBoxControl
		Public Sub New()
			CheckOnClick = True
			SelectionMode = System.Windows.Forms.SelectionMode.None
		End Sub

		Protected Overrides Function CreatePainter() As BaseControlPainter
			Return New MyPainter()
		End Function

		Protected Overrides Sub SetItemCheckStateCore(ByVal index As Integer, ByVal value As System.Windows.Forms.CheckState)
			If value = System.Windows.Forms.CheckState.Checked Then
				UnCheckAll()
			End If
			MyBase.SetItemCheckStateCore(index, value)
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/MyRadioListBox/MyRadioListBox.cs
using System;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyRadioListBox : CheckedListBoxControl
    {
        public MyRadioListBox()
        {
            CheckOnClick = true;
            SelectionMode = System.Windows.Forms.SelectionMode.None;
        }

        protected override BaseControlPainter CreatePainter()
        {
            return new MyPainter();
        }

        protected override void SetItemCheckStateCore(int index, System.Windows.Forms.CheckState value)
        {
            if (value == System.Windows.Forms.CheckState.Checked)
                UnCheckAll();
            base.SetItemCheckStateCore(index, value);
        }
    }
}
